src/OHIFStandaloneViewer.js

```diff
@@ -79,24 +87,84 @@ class OHIFStandaloneViewer extends Component {
       );
     }
 
+    /**
+     * Note: this approach for routing is caused by the conflict between
+     * react-transition-group and react-router's <Switch> component.
+     *
+     * See http://reactcommunity.org/react-transition-group/with-react-router/
+     */
+    const routes = [
+      {
+        path: '/',
+        Component: StudyListRouting,
+      },
+      {
+        path: '/studylist',
+        Component: StudyListRouting,
+      },
+      {
+        path: '/viewer',
+        Component: StandaloneRouting,
+      },
+      {
+        path: '/viewer/:studyInstanceUids',
+        Component: ViewerRouting,
+      },
+      {
+        path: '/study/:studyInstanceUid/series/:seriesInstanceUids',
+        Component: ViewerRouting,
+      },
+      {
+        path: '/IHEInvokeImageDisplay',
+        Component: IHEInvokeImageDisplay,
+      },
+    ];
+
+    const currentPath = this.props.location.pathname;
+    const noMatchingRoutes = !routes.find(r =>
+      matchPath(currentPath, {
+        path: r.path,
+        exact: true,
+      })
+    );
+
     return (
-      <Switch>
+      <>
+        <NProgress isAnimating={this.state.isLoading}>
+          {({ isFinished, progress, animationDuration }) => (
+            <Container
+              isFinished={isFinished}
+              animationDuration={animationDuration}
+            >
+              <Bar progress={progress} animationDuration={animationDuration} />
+            </Container>
+          )}
+        </NProgress>
```

Comment:
If we don't intend to reuse bar or container, it may make more sense to create a single component out of this and kill those components.